### DISCUSSION TITLE
Trap all kinds of errors, IIS through MVC

### DISCUSSION THREAD

**Discussion by alexandrejobin** [Dotnet-Boxed/Templates#27] [Q&A]
hi!

it would be great to have a way to trap all errors, from IIS through MVC. A demo project that you can find [here](https://github.com/stevejay/Mvc5ErrorHandling) explain how you can do that.

I have experimented a little bit with this solution but haven't had the time to finish it. Maybe you can do something good for your own boilerplate project!

**Comment by RehanSaeed** [comment]
I had a quick read, there doesn't seem to be any documentation explaining what it's doing to handle the errors and the link to the blog posts don't work. I had a quick look at the code anyway and it seems to be using a IHttpModule to intercept errors.

What errors do you want handled that aren't already being handled? MVC 5 Boilerplate does not cover every HTTP error status code in the httpErrors section of web.config but covers all the main ones. I think error handling in MVC 5 is a total mess and they've improved things a lot in MVC 6.

**Comment by alexandrejobin** [comment]
His blog seems to be down but you can play with his demo project. It show you how you can trap IIS errors and asp.net errors. Just change de settings at the top and you have all kinds of error that you can generate. The goal would be to trap every kind of errors and not just the mvc ones.

I don't know about MVC6 so i can't comment on that one.

**Comment by RehanSaeed** [comment]
@skoub Thanks for raising the issue :+1: . It would be nice to implement better error handling in MVC 5 as it's one of the annoying things about IIS/ASP.NET 4.6. I have limited time/resources and lots to do with MVC 6 as that is inevitably the future. If this was a problem with MVC 6 too then I think I'd spend more time on it but right now, I'm going to close this issue.

**Comment by alexandrejobin** [comment]
I'm working on it when i have free time. If i can finish it, i will post it there.
